mlx_audio/tts/models/llama/llama.py

```diff
@@ -297,8 +337,20 @@ def generate(
         **kwargs,
     ):
         prompt = text.replace("\\n", "\n").replace("\\t", "\t")
+
+        # Split by newlines first if needed
         prompts = prompt.split(split_pattern)
-        prompts = [f"{voice}: " + p for p in prompts]
+
+        # Further split long prompts into smaller chunks
+        all_prompts = []
+        for p in prompts:
+            if len(p) > 300:  # Only split if text is longer than 300 chars
+                chunks = self._split_text_into_chunks(p)
+                all_prompts.extend(chunks)
+            else:
+                all_prompts.append(p)
+
+        prompts = [f"{voice}: " + p for p in all_prompts if p.strip()]
```

Comment:
On second thought, I think we should revert this most changes except this part.

Because the downstream code handled prompt list.